django/test/html.py

```diff
@@ -96,6 +97,19 @@ def _count(self, element, count=True):
                     elif element in child:
                         return 1
             else:
+                # Look for a complete sequence of element's children.
+                # Refs #27906.
+                if isinstance(element, RootElement) and element.children:
+                    elem_child = element.children[elem_child_idx]
+                    # start or continue match
+                    if elem_child == child:
+                        elem_child_idx += 1
+                        # complete match
+                        if elem_child_idx == len(element.children):
+                            i += 1
+                            elem_child_idx = 0
```

Comment:
We're missing tests here. I'd expect a failure with this diff: 

```diff
diff --git a/django/test/html.py b/django/test/html.py
index b2af0938ce..8c2543a5da 100644
--- a/django/test/html.py
+++ b/django/test/html.py
@@ -86,7 +86,6 @@ class Element:
             if self.children == element.children:
                 return 1
         i = 0
-        elem_child_idx = 0
         for child in self.children:
             # child is text content and element is also text content, then
             # make a simple "text" in "text"
@@ -100,16 +99,10 @@ class Element:
                 # Look for a complete sequence of element's children.
                 # Refs #27906.
                 if isinstance(element, RootElement) and element.children:
-                    elem_child = element.children[elem_child_idx]
+                    elem_child = element.children[0]
                     # start or continue match
                     if elem_child == child:
-                        elem_child_idx += 1
-                        # complete match
-                        if elem_child_idx == len(element.children):
-                            i += 1
-                            elem_child_idx = 0
-                    else:
-                        elem_child_idx = 0
+                        i += 1
                 i += child._count(element, count=count)
                 if not count and i:
                     return i
```

(c.f. _"Better comments..."_ above)